tests/kafkatest/tests/produce_consume_validate.py

```diff
@@ -59,14 +59,18 @@ def stop_producer_and_consumer(self):
 
     def run_produce_consume_validate(self, core_test_action=None, *args):
         """Top-level template for simple produce/consume/validate tests."""
+        try:
+            self.start_producer_and_consumer()
 
-        self.start_producer_and_consumer()
+            if core_test_action is not None:
+                core_test_action(*args)
 
-        if core_test_action is not None:
-            core_test_action(*args)
-
-        self.stop_producer_and_consumer()
-        self.validate()
+            self.stop_producer_and_consumer()
+            self.validate()
+        except BaseException as e:
+            for s in self.test_context.services:
+                self.mark_for_collect(s)
```

Comment:
This seems a bit dangerous since `mark_for_collect` with only a `Service` parameter will collect all logs. Some `Service` classes can generate very large files which are listed in their `logs`, but even upon failure I'm not sure it's always reliable to grab them all. For example, verifiable producer and consumer log a *lot* to stdout, and we have to grab that data for analysis, but rarely include it in the test output.

I'm not sure those are the best examples since they are probably required for debugging pretty much any time they are used and the test fails, but I'm worried about similar cases. Or maybe this shouldn't be an issue since tests shouldn't ever be failing consistently (:pray:) , so a few large datasets from failed tests won't be an issue?